query-algebrizer/src/clauses/pattern.rs

```diff
@@ -238,8 +238,29 @@ impl ConjoiningClauses {
                 if value_type.is_none() {
                     self.wheres.add_intersection(ColumnConstraint::HasType(col.clone(), typed_value_type));
                 }
+            },
+        }
 
+        match pattern.tx {
+            PatternNonValuePlace::Placeholder => (),
+            PatternNonValuePlace::Variable(ref v) => {
+                self.bind_column_to_var(schema, col.clone(), DatomsColumn::Tx, v.clone());
+            },
+            PatternNonValuePlace::Entid(entid) => {
+                // TODO: we want to check whether the tx-id is within range for the database's tx partition. 
+                // (That applies after ident lookup, too.)
+                // Possible solution: https://github.com/mozilla/mentat/tree/fluffyemily/tx-id-check
```

Comment:
I had a go at achieving this:
https://github.com/mozilla/mentat/compare/fluffyemily/tx-places...fluffyemily/tx-id-check

It feels messy but I couldn't find another way of getting access to the `tx` partition map inside the query callstack.